### PR TITLE
lp1739734: Fixed federated SE regression introduced in the gcc7 fixes. (5.5)

### PR DESCRIPTION
**Pull request by dutow**
Also added a related test.

### CODE REVIEW

File: mysql-test/suite/federated/federated_bug_lp1739734.test [hunk 1]
```diff
@@ -0,0 +1,53 @@
+--echo #
```
Comment (laurynas-biveinis) [inline mysql-test/suite/federated/federated_bug_lp1739734.test]: Please check the Confluence page on the testcase naming, it should be named percona_bug1739734

Comment (laurynas-biveinis) [inline mysql-test/suite/federated/federated_bug_lp1739734.test]: No .result file

File: mysql-test/suite/federated/federated_bug_lp1739734.test [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+--echo # This is a bug introduced in the artful/gcc7 compilation fixes, caused by
+--echo # a moved break statement.
+
+--source suite/federated/include/federated.inc
+
+connection master;
+CREATE DATABASE lp1739734;
+use lp1739734;
+
+DROP SERVER IF EXISTS local_server;
```
Comment (laurynas-biveinis) [inline mysql-test/suite/federated/federated_bug_lp1739734.test]: No need for DROP IF EXISTS in the testcases

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

**Comment by dutow** [comment]
https://jenkins.percona.com/job/percona-server-5.5-param/1611/

**Review by laurynas-biveinis — APPROVED** [review]

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Pull request merged**